### ISSUE TITLE
drawing tools - live vs. object

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/mobilise#507]
this issue is to work out the nuances of the drawing tools, now that it seems we need to have 2 different types of drawing tool:
- live drawing (more or less how it worked in the old UpStage - a pen & tools to adjust size, colour, opacity; you draw on the stage and everyone sees the drawing appearing as you draw it).
- object drawing (more or less how it's currently working in the new UpStage - you create a drawing and it becomes an object, that can be saved and manipulated on the stage like an avatar)

QUESTIONS:
- should there be one icon in the left-hand tool bar for drawing, which then opens a palette which has a switch to toggle between the two different kinds of drawing?
- or should there be two different icons in the left-hand tool bar, with each kind of drawing having its own overlay/palette?
- "live drawing" is clear what it is, but what should the other drawing tool be called? e.g. object drawing, JIT drawing, avatar drawing ... something else?
- or name them as different modes, e.g. "drawing board" and "drawing object"
- how can we make sure that users don't get confused between the different modes, e.g. they think they're making a drawing object but they're in the live mode?
- what icons can we have? the pen icon is currently in the left-hand tool bar

**Comment by elisadcg** [comment]
i propose this kind of icon, a mix between pen tool and live stream
![Screenshot_20210803_111923](https://user-images.githubusercontent.com/7611913/127991286-0d214119-2a07-4538-afa8-f3afd2b9defb.png)
2 tool or 1 tool... i hesitate...
It's faster to have 2 differents tool during cyberformance because you don't have to activate drawing tool and after activate the live before just draw.
And 2 differents tool suggest 2 differents result in performance.

It's more or less like the difference between props and avatar at the end :D 

So ok i'm more in favour 2 separate tool.
But for the name, i let the english master ^^

**Comment by helenvarley** [comment]
yes i'm leaning towards 2 separate tools ... would it work to have the pen icon that we have now for the object one, & the one elisa has proposed here for the live drawing? 
but when it's small & in the tool bar, will it be easy to confuse with the stream tool?

**Comment by elisadcg** [comment]
![507](https://user-images.githubusercontent.com/7611913/128017757-76268505-2c05-4803-bce4-be120c259bf7.png)
 i change again the icon for live drawing

do you think it very similar to streaming ?

**Comment by nguyenhongphat0** [comment]
As a first step, I removed the Live Drawing toggle and now we have 2 separated Save button: Save as Avatar or Save as Prop. Do you think we need a separated icon for them @elisadcg 🤔 ?
<img width="906" alt="Screenshot 2021-08-03 at 23 52 09" src="https://user-images.githubusercontent.com/29925961/128055178-850fb499-e937-477a-abc4-3432cad3dc19.png">

**Comment by vickis** [comment]
Great work Phat!
So the different save take on the functions and parameters of either avatar
or prop (eg can or can’t speak)
And the prop object (drawing) can be attached to the avatar (drawing)

**Comment by vickis** [comment]
I liked the reference to the live (with the stream)
  And it made visual sense to me

but if it might be confusing perhaps live is just the pencil for draw live
(drawing board)
And could icon for draw object a shape
Perhaps cube or square with dashed line ...
 or circle ⭕️

The drawing object can be saved and orientation can change
Drawing board is live (the current draw icon has a sense of potential
movement :)

**Comment by helenvarley** [comment]
when i save as avatar, does it mean that the drawing would then appear in the avatar overlay as another avatar? & the same with prop - it's then available in the prop overlay? & would it be available for all players, or only me?

**Comment by helenvarley** [comment]
for the icons - the live one that looks like streaming is too similar to streaming.

**Comment by elisadcg** [comment]
> As a first step, I removed the Live Drawing toggle and now we have 2 separated Save button: Save as Avatar or Save as Prop. Do you think we need a separated icon for them @elisadcg thinking ?
> <img alt="Screenshot 2021-08-03 at 23 52 09" width="906" src="https://user-images.githubusercontent.com/29925961/128055178-850fb499-e937-477a-abc4-3432cad3dc19.png">

the icon could be save button + prop button or save button + avatar button ?

**Comment by nguyenhongphat0** [comment]
> when i save as avatar, does it mean that the drawing would then appear in the avatar overlay as another avatar? & the same with prop - it's then available in the prop overlay? & would it be available for all players, or only me?

No Helen, drawing is still drawing, they will be shown on the Draw tools. And your saved drawings will only available for you only! And sorry for the late response 😅

**Comment by helenvarley** [comment]
no problem! :)

**Comment by nguyenhongphat0** [comment]
I believe this issue is resolved 🙌 If you would need any more enhancements on the live drawing, just open a new issue for it 👌

**Comment by vickis** [comment]
Thanks Phat! :D
